classes/monograph/ChapterDAO.php

```diff
@@ -114,24 +114,44 @@ public function getByDoiId(int $doiId): DAOResultFactory
     }
 
     /**
-     * Retrieve all chapters by source chapter id.
+     * Retrieve all chapters by source chapter ID or retrieve a chapter by source chapter ID and publication ID.
      *
+     * @param int $sourceChapterId Chapter ID from first version of this chapter
+     * @param bool $orderByPublicationId optional, default is true
+     * @param ?int $publicationId optional, changes return value to Chapter or null
      *
+     * @return DAOResultFactory|Chapter|null Chapter|null only with publication ID
      */
-    public function getBySourceChapterId(int $sourceChapterId, bool $orderByPublicationId = true): DAOResultFactory
+    public function getBySourceChapterId(int $sourceChapterId, bool $orderByPublicationId = true, ?int $publicationId = null): DAOResultFactory|Chapter|null
```

Comment:
Rather than forcing `getBySourceChapterId` to behave in two different ways with different types of returns, I'd suggest either
- writing a separate function that gets a `Chapter` when given both a `$sourceChapterId` and a `$publication`, or
- Having this function consistently return a `DAOResultFactory`, even if calling code supplies a `$publicationId` and only expects a single matching return at most.

We'll be rewriting this code (along with the rest of the software, gradually) to use the new `EntityDAO` toolset (see e.g. https://github.com/pkp/pkp-lib/issues/7129). The fewer personalities that functions like this have, the easier it'll be to refactor.

Reply:
(I've already merged your citationStyleLanguage PR, so a change to this would also mean a new small PR to tweak the plugin. Sorry!)

Reply:
Well, I've split this in two functions. The new citationStyleLanguage PR you can find here: [https://github.com/pkp/citationStyleLanguage/pull/102](https://github.com/pkp/citationStyleLanguage/pull/102)